clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1515,6 +1560,17 @@ protected void processCommonConsumerAction(
         });
 
         long consumptionStatePrepTimeStart = System.currentTimeMillis();
+        if (!checkDatabaseIntegrity(partition, topic, offsetRecord, newPartitionConsumptionState)) {
+          LOGGER.warn(
+              "Restart ingestion from the beginning by resetting OffsetRecord for topic: {} and partition: {}",
+              topic,
+              partition);
+          resetOffset(partition, topicPartition, true);
+          newPartitionConsumptionState = partitionConsumptionStateMap.get(partition);
+          offsetRecord = newPartitionConsumptionState.getOffsetRecord();
+        }
+
+        newPartitionConsumptionState.setLeaderFollowerState(leaderState);
```

Comment:
Should this line be moved to the above `if` scope?

Reply:
Moved it.